### PR TITLE
REL-522364 - Converted C# API calls to REST in RelativityVersionHelpe…

### PR DESCRIPTION
**Pull request by aarongilbert123**
…r.cs

### CODE REVIEW

File: CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/ConfirmInstallerAndInstanceRelativityVersionAreEqualModule.cs [hunk 1]
```diff
@@ -66,13 +66,8 @@ protected override void ProcessRecordCode()
...
 
-			IConnectionHelper connectionHelper = new ConnectionHelper(
-				relativityInstanceName: RelativityInstanceName,
-				relativityAdminUserName: RelativityAdminUserName,
-				relativityAdminPassword: RelativityAdminPassword,
-				sqlAdminUserName: SqlAdminUserName,
-				sqlAdminPassword: SqlAdminPassword);
-			IRelativityVersionHelper relativityVersionHelper = new RelativityVersionHelper(connectionHelper);
+			IRestHelper restHelper = new RestHelper();
+			IRelativityVersionHelper relativityVersionHelper = new RelativityVersionHelper(restHelper, RelativityInstanceName, RelativityAdminUserName, RelativityAdminPassword);
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/ConfirmInstallerAndInstanceRelativityVersionAreEqualModule.cs:70]: Why are we not using the IConnectionHelper and passing the credentials directly to the RelativityVersionHelper constructor? Using IConnectionHelper helps with Dependency Injection.

- [ ] Use IConnectionHelper

File: CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs [hunk 2]
```diff
@@ -11,29 +12,31 @@ namespace Helpers.Implementations
...
+			AdminPassword = adminPassword;
+			RestHelper = restHelper;
 		}
 
-		public void ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion)
+		public async Task ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion)
 		{
-			try
+			HttpClient httpClient = RestHelper.GetHttpClient(InstanceAddress, AdminUsername, AdminPassword);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.InstanceDetailsService.EndpointUrl, "");
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs:31]: Doesn't this should be a GET call as we are retrieving data?

- [ ] Use GET

File: CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs [hunk 3]
```diff
@@ -11,29 +12,31 @@ namespace Helpers.Implementations
...
 		{
-			ServiceFactory = connectionHelper.GetServiceFactory();
+			InstanceAddress = instanceAddress;
+			AdminUsername = adminUsername;
+			AdminPassword = adminPassword;
+			RestHelper = restHelper;
 		}
 
-		public void ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion)
+		public async Task ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs:28]: Do we also have tests for this class?

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs (hunk 2 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs:31]: You use POST for this endpoint, as listed in this page: https://platform.relativity.com/9.5/Content/REST_API/Resource_Tasks/Retrieve_the_version_of_a_Relativity_instance.htm

File: CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs (hunk 3 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/RelativityVersionHelper.cs:28]: Yes we have integration tests

File: CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/ConfirmInstallerAndInstanceRelativityVersionAreEqualModule.cs (hunk 1 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/ConfirmInstallerAndInstanceRelativityVersionAreEqualModule.cs]: Done

**Review by calimetikcura — APPROVED** [review]

File: CSharp/DevVmPowershell/Helpers/Interfaces/IRelativityVersionHelper.cs [hunk 4]
```diff
@@ -8,6 +8,6 @@ namespace Helpers.Interfaces
 {
 	public interface IRelativityVersionHelper
 	{
-		void ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion);
+		Task ConfirmInstallerAndInstanceRelativityVersionAreEqual(string installerRelativityVersion);
```
Comment (TommyBigNose) [inline CSharp/DevVmPowershell/Helpers/Interfaces/IRelativityVersionHelper.cs:11]: Don't want to be that guy, but should this function now end in Async?

**Review by TommyBigNose — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Interfaces/IRelativityVersionHelper.cs (hunk 4 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Interfaces/IRelativityVersionHelper.cs]: Done

**Review by TommyBigNose — APPROVED** [review]

**Review by calimetikcura — APPROVED** [review]

**Pull request merged**